### PR TITLE
Add support for date range values to `Calendar`

### PR DESCRIPTION
**Pull request by dwick**
This PR adds support for date range values to the calendar widget.

Example:

```js
var start = new Date(2015, 0, 1)
   ,  end = new Date(2015, 0, 10);

<Calendar value={[start, end]} />
```

In the above example, `start` and/or `end` can be `null` as well.  If both values are `null`, then nothing is selected, and if a single value is `null` then only a single date is selected.

Let me know if you need anything else from me to get this in.

### CODE REVIEW

**Comment by dwick** [comment]
not quite ready.  i'll reopen in a bit.

**Comment by jquense** [comment]
ya can leave it open and just keep committing to the PR if that's easier for you. I'm excited to try this out, I'm not quite sure what the value is, but i'm open to finding ou :)

**Comment by dwick** [comment]
@jquense I'm having a bit of trouble reasoning about the `controlledInput` mixin for this and as a result `Calendar#change` is a bit hairy. I'm also thinking drag events might be needed to do this correctly.  I've updated `example/example.jsx` with something that sort of works so you can what I'm talking about.

So far I'm pretty happy with the changes to `Month`, `Year`, `Decade`, and `Century`, but `Calendar` is pretty broken.

Would love your thoughts, not really happy with where this is headed.

**Comment by dwick** [comment]
I think my use case might be too specific, closing unless you find this useful.

**Comment by jquense** [comment]
hey @dwick this is cool. I think ultimately you've got a good idea here, selecting ranges of dates would be a nice feature for the Calendar. The big challenge here is getting the interaction model right. Here are some thoughts on that, if you'd like to take it on, if not we can file this away and it'll be a great starting point down the road if I get some time. Either way I appreciate you taking the time to write this up.

- the click to select model doesn't work great here (your hairy onChange). I think (like you noted) you'd need to do proper drag to select. That poses some additional issues, like needing to shift months as you dragged past the start or end of the month (think Jan 16 to Feb 5th).
- You'd also need to implement the same functionality in a keyboard only model (like ctrl + arrow key) to accommodate users that don't use a mouse.

File: src/Calendar.jsx [hunk 1]
```diff
@@ -77,12 +80,12 @@ var Calendar = React.createClass({
...
   getInitialState: function(){
-    var value = this.inRangeValue(this.props.value);
+    var value = this.normalizeValue(this.props.value);
 
     return {
       selectedIndex: 0,
-      view:          this.props.initialView || 'month',
-      currentDate:   value ? new Date(value) : new Date()
+      view: this.props.initialView || 'month',
+      currentDateRange: value ? value : [new Date(), null]
```
Comment (jquense) [inline src/Calendar.jsx]: `currentDate` determines what month/year/decade to show in any of the views, so it should always just be a single date. even if the value is a range

**Pull request closed without merging**